src/index.rs

```diff
@@ -68,95 +93,42 @@ impl ResourceIndex {
-                UNIX_EPOCH.add(Duration::from_millis(
-                    str.parse().map_err(|_| ArklibError::Parse)?,
-                ))
-            };
-
-            let id = {
-                let str = parts.next().ok_or(ArklibError::Parse)?;
-                ResourceId::from_str(str)?
-            };
-
-            let path: String =
-                itertools::Itertools::intersperse(parts, " ").collect();
-            let path: PathBuf = root_path.join(Path::new(&path));
-            match CanonicalPathBuf::canonicalize(&path) {
-                Ok(path) => {
-                    log::trace!("[load] {} -> {}", id, path.display());
-                    index.insert_entry(path, IndexEntry { id, modified });
-                }
-                Err(_) => {
-                    log::warn!("File {} not found", path.display());
-                    continue;
-                }
-            }
-        }
+        let file = File::open(index_path)?;
+        let reader = BufReader::new(file);
+        let index: ResourceIndex = serde_json::from_reader(reader)?;
 
         Ok(index)
     }
 
+    /// Stores the resource index to the file system
+    ///
+    /// This function writes the index to the file system. It writes the index
+    /// to `root_path/.ark/index` and creates the directory if it does not exist
     pub fn store(&self) -> Result<()> {
         log::info!("Storing the index to file");
-
         let start = SystemTime::now();
 
-        let index_path = self
-            .root
-            .to_owned()
-            .join(ARK_FOLDER)
-            .join(INDEX_PATH);
-
-        let ark_dir = index_path.parent().unwrap();
-        fs::create_dir_all(ark_dir)?;
-
-        let mut file = File::create(index_path)?;
-
-        let mut path2id: Vec<(&CanonicalPathBuf, &IndexEntry)> =
-            self.path2id.iter().collect();
-        path2id.sort_by_key(|(_, entry)| *entry);
-
-        for (path, entry) in path2id.iter() {
-            log::trace!("[store] {} by path {}", entry.id, path.display());
-
-            let timestamp = entry
-                .modified
-                .duration_since(UNIX_EPOCH)
-                .map_err(|_| {
-                    ArklibError::Other(anyhow!("Error using duration since"))
-                })?
-                .as_millis();
-
-            let path =
-                pathdiff::diff_paths(path.to_str().unwrap(), self.root.clone())
-                    .ok_or(ArklibError::Path(
-                        "Couldn't calculate path diff".into(),
-                    ))?;
-
-            writeln!(file, "{} {} {}", timestamp, entry.id, path.display())?;
+        let ark_folder = self.root.join(ARK_FOLDER);
+        if !ark_folder.exists() {
+            fs::create_dir(ark_folder.clone())?;
         }
+        let index_path = ark_folder.join(INDEX_PATH);
+        let file = File::create(index_path)?;
+
+        let writer = BufWriter::new(file);
+        serde_json::to_writer(writer, self)?;
```

Comment:
The file format from `main` branch is also sorted by timestamps. This is useful for keeping the index file same on different devices and between runs, and also convenient for debugging.

Reply:
Yes. good point
we can sort before serializing.